reaper_test.go

```diff
@@ -102,21 +84,26 @@ func createContainerRequest(customize func(ContainerRequest) ContainerRequest) C
 
 	req.Labels[core.LabelReaper] = "true"
 	req.Labels[core.LabelRyuk] = "true"
+	delete(req.Labels, core.LabelReap)
 
-	if customize == nil {
-		return req
+	for _, customize := range customize {
+		customize(&req)
 	}
 
-	return customize(req)
+	return req
 }
 
-func TestContainerStartsWithoutTheReaper(t *testing.T) {
-	config.Reset() // reset the config using the internal method to avoid the sync.Once
-	tcConfig := config.Read()
-	if !tcConfig.RyukDisabled {
-		t.Skip("Ryuk is enabled, skipping test")
```

Comment:
question: We have a dedicated pipeline that runs all the tests for the code with the Reaper disabled, so there are tests that we skip in there. For that, we set the env var in the GH workflow: please see ci-test-go.yaml file:

```yaml
TESTCONTAINERS_RYUK_DISABLED: "${{ inputs.ryuk-disabled }}"
```

Do we still need to skip those tests?

Reply:
With these changes a single test run can validate with and without the reaper being abled, so there's no need to skip tests anymore, see `func TestContainer`.

There might still benefit of a full test run with the reaper disabled to catch issues in modules, but we could consider removing that as it would cut the number of tests in half, reducing the time to validate. Something to discuss and include in a separate PR.

Reply:
Yeah, our reaper-off pipeline runs only for the core. Not sure if it would be desired to run the modules without the reaper as a second step in the same pipeline 🤔

Reply:
The compose module is the only one which springs to mind, it has a `TestDockerComposeAPI_WithoutReaper` and `TestDockerComposeAPI_WithReaper`.